accepted/file-enumeration.md

```diff
@@ -0,0 +1,238 @@
+# Extensible File Enumeration
+
+**PM** [Immo Landwerth](https://github.com/terrajobst) | **Dev** [Jeremy Kuhne](https://github.com/jeremykuhne)
+
+Enumerating files in .NET provides limited configurability. You can specify a simple DOS style pattern and whether or not to look recursively. More complicated filtering requires post filtering all results which can introduce a significant performance drain.
+
+Recursive enumeration is also problematic in that there is no way to handle error states such as access issues or cycles created by links.
+
+These restrictions have a significant impact on file system intensive applications, a key example being MSBuild. We can address these restrictions and provide performant, configurable enumeration.
+
+## Scenarios and User Experience
+
+1. MSBuild can custom filter filesystem entries with limited allocations and form the results in any desired format.
+2. Users can build custom enumerations utilizing completely custom or provided commonly used filters and transforms.
+
+## Requirements
+
+1. Filtering based on common file system data is possible
+	a. Name
+	b. Attributes
+	c. Time stamps
+    d. File size
```

Comment:
> d. File size [](start = 3, length = 13)

What doesn't pop here is how the filter an be provided. I think you want to call out explicitly that the filter is a customer function and thus supports all sorts of filtering criteria, including calling custom functions and arbitrary combinators (AND, OR, NOT etc).